### PR TITLE
MAPD-363  added changes for street-view directive build

### PR DESCRIPTION
**Pull request by realtyjustin**
Added some config and a command hook for isolating street-view directive in distribution.  Also accounting for it within other grunt tasks where seemingly pertinent (please check).

### CODE REVIEW

File: grunt/options.coffee [hunk 1]
```diff
@@ -56,15 +60,26 @@ concatDist =
   ]
   dest: "dist/<%= pkg.name %>.js"
 
-clone = (obj) ->
-  _.extend {}, obj
-
 concatDistMapped = clone concatDist
 concatDistMapped.options = _.extend clone(concatDist.options),
```
Comment (nmccready) [inline grunt/options.coffee]: use cloneDeep here

**Comment by nmccready** [comment]
Looks good besides the one comment.   Also can you merge latest into this?

**Comment by nmccready** [comment]
@realtyjustin

**Comment by realtyjustin** [comment]
@nmccready ok - I was wondering about that 'clone' that was there... so I went ahead and removed all of those in favor of lodash cloneDeep

**Comment by nmccready** [comment]
Yeah thats fine there is just an old reference to clone still in the code. I did it like that if we ever moved to underscore as there is no cloneDeep. Anyway im fine with it. However I still can not merge.

**Comment by nmccready** [comment]
@realtyjustin conflicts need to be resolved.

**Comment by realtyjustin** [comment]
@nmccready  conflicts should be cleaned up now

File: dist/angular-google-maps-street-view.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  angular.module('uiGmapgoogle-maps.directives.api.options.builders', []);
+
+  angular.module('uiGmapgoogle-maps.directives.api.models.child', ['uiGmapgoogle-maps.directives.api.utils', 'uiGmapgoogle-maps.directives.api.options', 'uiGmapgoogle-maps.directives.api.options.builders']);
+
+  angular.module('uiGmapgoogle-maps.directives.api.models.parent', ['uiGmapgoogle-maps.directives.api.managers', 'uiGmapgoogle-maps.directives.api.models.child', 'uiGmapgoogle-maps.providers']);
+
+  angular.module('uiGmapgoogle-maps.directives.api', ['uiGmapgoogle-maps.directives.api.models.parent']);
+
+  angular.module('uiGmapgoogle-maps', ['uiGmapgoogle-maps.directives.api', 'uiGmapgoogle-maps.providers']);
+
```
Comment (nmccready) [inline dist/angular-google-maps-street-view.js]: So only the module was loaded. the directive is not in here nor are it's dependencies. @realtyjustin

File: grunt/options.coffee [hunk 3]
```diff
@@ -56,15 +57,25 @@ concatDist =
...
-  sourceMapName: "dist/<%= pkg.name %>_dev_mapped.js.map"
+concatDistMapped = _.cloneDeep concatDist
+concatDistMapped.options.sourceMap = true
+concatDistMapped.options.sourceMap = "dist/<%= pkg.name %>_dev_mapped.js.map"
 concatDistMapped.dest = "dist/<%= pkg.name %>_dev_mapped.js"
 
+# configs for concat street view directive & dev mapping
+concatStreetView = _.cloneDeep concatDist
+concatStreetView.src = [
+  "src/coffee/module"
```
Comment (nmccready) [inline grunt/options.coffee]: You don't concat coffee files, you concat js files which are in /tmp.

**Comment by nmccready** [comment]
There are problems with the code itself, sorry I should have caught this earlier.

**Comment by realtyjustin** [comment]
@nmccready  can you double check - there is already a ".map( (f) -> "tmp/#{f}.js" )" tagged on that list.

**Comment by nmccready** [comment]
@realtyjustin ahh ok the `.map` is correct, so why is it not in the dist?

**Comment by realtyjustin** [comment]
@nmccready ok - I inadvertently left the .coffee extension on that file.  try this next commit.

File: dist/angular-google-maps-street-view.js [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+ */
+
+
+/*
+StreetViewPanorama Directive to care of basic initialization of StreetViewPanorama
+ */
+
+(function() {
+  angular.module('uiGmapgoogle-maps').directive('uiGmapStreetViewPanorama', [
+    'uiGmapGoogleMapApi', 'uiGmapLogger', 'uiGmapGmapUtil', 'uiGmapEventsHelper', function(GoogleMapApi, $log, GmapUtil, EventsHelper) {
```
Comment (nmccready) [inline dist/angular-google-maps-street-view.js]: Almost there, you need the dependencies files as well 

```
'uiGmapGoogleMapApi', 'uiGmapLogger', 'uiGmapGmapUtil', 'uiGmapEventsHelper'
```

**Comment by nmccready** [comment]
@realtyjustin just add the dependencies b4 panorama and we're set

**Comment by realtyjustin** [comment]
@nmccready ok - try that one

**Comment by nmccready** [comment]
Lgtm

**Pull request merged**